### PR TITLE
feat(auto-catalog): add hideHeading and indexType

### PR DESCRIPTION
**Pull request by zfben**
This pull request work for https://github.com/vuepress-theme-hope/vuepress-theme-hope/issues/3333

Add hideHeading and indexType.

### CODE REVIEW

File: docs/auto-catalog/src/config.md [hunk 1]
```diff
@@ -168,3 +168,10 @@ Max level of catalog.
 - Default: `false`
 
 Whether display index number for catalog.
+
+### hideCategory
```
Comment (Mister-Hope) [inline docs/auto-catalog/src/config.md:172]: It's better to be `hideHeading` or `hideTitle`

File: packages/auto-catalog/src/client/components/AutoCatalog.ts [hunk 2]
```diff
@@ -249,7 +260,7 @@ export default defineComponent({
                 ),
                 children.length
                   ? h(
-                      "ul",
+                      props.index ? "ol" : "ul",
```
Comment (Mister-Hope) [inline packages/auto-catalog/src/client/components/AutoCatalog.ts:263]: Are nesting ol markers be `1.1` `1.1.1` in all kinds of devices? I think not, the styles needs to be handled correctly.

**Review by Mister-Hope — CHANGES_REQUESTED** [review]
Fix those. Also zh docs are missing

**Comment by Mister-Hope** [comment]
Is this PR still under development? @zfben

**Comment by zfben** [comment]
> Is this PR still under development? @zfben

I'm busy these days, I will do it later.

**Comment by zfben** [comment]
@Mister-Hope I have updated code and add indexType to let users can choose ul or ol by themselves.

**Comment by Mister-Hope** [comment]
Sorry but  I cleaned up all the deps PR at main, could.you make a rebase?

**Comment by zfben** [comment]
@Mister-Hope done

**Pull request merged**